flink-connector-oceanbase-cdc/src/main/java/com/ververica/cdc/connectors/oceanbase/source/OceanBaseRichParallelSourceFunction.java

```diff
@@ -0,0 +1,518 @@
+                                    latch.countDown();
+                                }
+                                break;
+                            case INSERT:
+                            case UPDATE:
+                            case DELETE:
+                                if (!started) {
+                                    break;
+                                }
+                                logMessageBuffer.add(message);
+                                break;
+                            case COMMIT:
+                                // flush buffer after snapshot completed
+                                if (!shouldReadSnapshot() || snapshotCompleted.get()) {
+                                    logMessageBuffer.forEach(
+                                            msg -> {
+                                                try {
+                                                    deserializer.deserialize(
+                                                            getRecordFromLogMessage(msg),
+                                                            outputCollector);
+                                                } catch (Exception e) {
+                                                    throw new FlinkRuntimeException(e);
+                                                }
+                                            });
+                                    logMessageBuffer.clear();
+                                    resolvedTimestamp = Long.parseLong(message.getTimestamp());
+                                }
+                                break;
+                            case DDL:
+                                // TODO record ddl and remove expired table schema
+                                LOG.trace(
+                                        "Ddl: {}",
+                                        message.getFieldList().get(0).getValue().toString());
+                                break;
+                            default:
+                                throw new UnsupportedOperationException(
+                                        "Unsupported type: " + message.getOpt());
+                        }
+                    }
+
+                    @Override
+                    public void onException(LogProxyClientException e) {
+                        LOG.error("LogProxyClient exception", e);
+                        logProxyClient.stop();
+                    }
+                });
+
+        logProxyClient.start();
+        LOG.info("LogProxyClient started");
+        latch.await();
+        LOG.info("LogProxyClient packet processing started");
+    }
+
+    private SourceRecord getRecordFromLogMessage(LogMessage message) throws Exception {
+        String databaseName = message.getDbName().replace(tenantName + ".", "");
+        // TODO make topic name configurable
+        String topicName = getDefaultTopicName(tenantName, databaseName, message.getTableName());
+
```

Comment:
Could we obtain meta data  like `database_name`, `table_name`, `talent_name` and `op_ts` from `LogMessage`? we can support metadata column if it is, all flink cdc connector supports meta data column.

Reply:
Done.